disnake/poll.py

```diff
@@ -0,0 +1,426 @@
+# SPDX-License-Identifier: MIT
```

Comment:
I think it would be better to have two separate objects for what the user can construct vs what the api returns; there's a bunch of fields the user constructible version won't have populated until some magic event, and you made accessing them throw AttributeError

I think you'd agree that physically restricting what the user can do with their object is better than merely mentioning it in the docstring (which seem to often grow large enough a good part of it won't be seen unless scrolled down)

Reply:
I agree with this. Something similar to what I did in the [Onboarding PR](https://github.com/DisnakeDev/disnake/pull/1011/commits/85e3d33167d8eb66baa19b5d90f91fde83890fe0) could be implemented.

Reply:
I'm going to mark this conversation as resolved. If you still have concerns please re-review the PR. Ty 🙏